circuit/algorithms/src/bhp/hasher/hash_uncompressed.rs

```diff
@@ -155,9 +157,9 @@ impl<E: Environment, const NUM_WINDOWS: u8, const WINDOW_SIZE: u8> HashUncompres
                 match &sum {
                     Some((sum_x, sum_y)) => {
                         // Convert the accumulated sum into a point on the twisted Edwards curve.
-                        // TODO: If we know that sum_y is nonzero, then we can replace with `sum_x * sum_y.inverse()`.
-                        let edwards_x = sum_x / sum_y; // 5 constraints;
-                        Group::from_x_coordinate(edwards_x) // 3 constraints
+                        let edwards_x = sum_x * sum_y.inverse(); // 1 constraint (`sum_y` is never 0)
+                        let edwards_y = (sum_x - &one) / (sum_x + &one); // 1 constraint (`sum_x` is never -1)
```

Comment:
This should be `let edwards_y = (sum_x - &one).div_unchecked(sum_x + &one)`.

Reply:
Switched both to use `div_unchecked`.